### ISSUE TITLE
Azure Cognitive Search - Keyword search in long text - Not returning expected results

### ISSUE THREAD

**Issue by lex-mw-lab** [Azure/azure-sdk-for-python#32058]
Searching for entries in a keyword-field stored in an Azure Cognitive Search index not returning expected results when searching for them in a long text. Multi word tokens as 'microsoft azure' are not returned as a match when looking in the text "This text contains microsoft azure"

Working with Azure Cognitive Search with the Python SDK. Say I'm building a search index where each document in the index has a "name" field. The name field (which can consist of multiple words) only makes sense if I tokenize the entire name is one token, so I use the "keyword_v2" tokenizer when building the analyzer for this field.

```python
from azure.search.documents.indexes.models import CustomAnalyzer

# Define the custom analyzer for the Name field
name_analyzer = CustomAnalyzer(name="name_analyzer",tokenizer_name="keyword_v2",
    token_filters=["lowercase"])`

# Specify the index schema

fields = [
        SimpleField(name="key", type=SearchFieldDataType.String, key=True),       
        SearchableField(name="name", type=SearchFieldDataType.String, analyzer_name="name_analyzer", searchable=True)
    ]
```

This works as expected when I test the analyzer using the REST API. As an example I have the following indexed entries in the name-field: ['microsoft azure', 'amazon aws', 'google cloud']. The custom analyzer I set up correctly tokenizes each entry as one token and not as multiple tokens (ex. 'microsoft' and 'azure').

The problem occurs when I search for the stored names in a text.

```python
text_example = "This is a text containing microsoft azure."

results = search_client.search(search_text=text_example, include_total_count=True, select= ['name'], search_fields= ['name'], highlight_fields= 'name', query_type= "full")

print ('Total Documents Matching Query:', results.get_count())
for result in results:
    print(result)
```

I expect when I search for a name in the text_example it will return a hit on 'microsoft azure', but it doesn't. It returns empty. I suspect because I use the same custom analyzer as both the index analyzer and search analyzer, it will tokenize the entire text_example as one token, which is not in the index. So it returns nothing.

Can I resolve this problem of searching for multiple word tokens in a long text in an efficient way using Azure Cognitive Search ?

Basically, I want the equivalent of the following Python snippet but with Azure Cognitive Search Index:

```python
name_list = ['microsoft azure', 'amazon aws', 'google cloud']
text_example = "This is a text containing microsoft azure."

for name in name_list:
    if name in text_example:
        print(f"Match found: {name}) 
```

**Comment by pvaneck** [comment]
Hey, thanks for the feedback. I'll route this issue to the appropriate people.

**Comment by xiangyan99** [comment]
To help me understand better of your scenario, 

you said "This works as expected when I test the analyzer using the REST API. "

Could you clarify you used REST API to create index, index or docs or do the search when it works as expected?

**Comment by lex-mw-lab** [comment]
@xiangyan99 The REST API was only used to test the custom analyzer I created (https://learn.microsoft.com/en-us/rest/api/searchservice/test-analyzer), which worked as expected.

**Comment by xiangyan99** [comment]
Did you have a chance to call rest api directly or use other language SDKs?

It looks to me the service behavior.

**Comment by lex-mw-lab** [comment]
@xiangyan99 I did call the REST API directly to test the custom analyzer, but the search index itself was created and implemented with the Python SDK. I don't think my issue is a bug or something, I just think I'm missing something practically while implementing some search functionality.

**Comment by xiangyan99** [comment]
Thanks for the updates.

What search service does is to search `search_text` among your docs stored in the search service.

In your case, you stored

docs ['microsoft azure', 'amazon aws', 'google cloud'] in the service and looked for "This is a text containing microsoft azure." among the docs.

I feel it is fair to return "not found".

If you think the behavior is wrong, please reach out to the service team or submit a feature request.

**Comment by lex-mw-lab** [comment]
@xiangyan99 Thanks for the explanation. I thought I maybe missed something in this case. Is there a way in which I can search for 'microsoft azure' as one keyterm in the text 'This text contains microsoft azure' so it returns a hit only when the complete keyterm 'microsoft azure' is found in the text ?

**Comment by xiangyan99** [comment]
If I understand correctly, you need to store text 'This text contains microsoft azure' into the search service and call method client.search(search_text='microsoft azure').
